tests/failpoints/test_io_error.rs

```diff
@@ -439,3 +439,53 @@ fn test_file_allocate_error() {
     assert_eq!(engine.first_index(1).unwrap(), 1);
     assert_eq!(engine.last_index(1).unwrap(), 4);
 }
+
+#[test]
+fn test_start_with_file_allocate_error() {
```

Comment:
I don't know exactly what is tested. Because "whether the file is reused of not" is not checked at all. I suggest adding checks with `fs.file_count()`.

Reply:
Extra annotations have been supplemented and the case has been refined.

This case is meant to mock the scenario that Engine is restarted after it is failed to be opened firstly when generating stale files in fail.